### ISSUE TITLE
[no-title]

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#12188]
[no-title]<br>
<br>
https://ift.tt/qOCUJlw<br>
<br>
Deep Learning Systems<br>
<br>
<div><div><div>
<tbody>
<tr>
<td>8/27</td>
<td>1 - Introduction / Logistics</td>
<td>Kolter</td>
<td><a href="https://dlsyscourse.org/slides/intro.pdf">pdf</a></td>
<td><a href="https://youtu.be/ftP5HeOvsI0">YouTube</a></td>
</tr>
<tr>
<td>8/29</td>
<td>2 - ML Refresher / Softmax Regression</td>
<td>Kolter</td>
<td><a href="https://dlsyscourse.org/slides/2-softmax_regression.pdf">pdf</a></td>
<td><a href="https://youtu.be/MlivXhZFbNA">YouTube</a></td>
</tr>
<tr>
<td>9/3</td>
<td>3 - Manual Neural Networks / Backprop</td>
<td>Kolter</td>
<td><a href="https://dlsyscourse.org/slides/manual_neural_nets.pdf">pdf</a></td>
<td>
<a href="https://youtu.be/OyrqSYJs7NQ">YouTube (pt 1)</a> <a href="https://youtu.be/JLg1HkzDsKI">YouTube (pt 2)</a>
</td>
</tr>
<tr>
<td>9/5</td>
<td>4 - Automatic Differentiation</td>
<td>Chen</td>
<td><a href="https://dlsyscourse.org/slides/4-automatic-differentiation.pdf">pdf</a></td>
<td><a href="https://youtu.be/56WUlMEeAuA">YouTube</a></td>
</tr>
<tr>
<td>9/10</td>
<td>5 - Automatic Differentiation Implementation</td>
<td>Chen</td>
<td><a href="https://github.com/dlsyscourse/lecture5/blob/main/5_automatic_differentiation_implementation.ipynb">ipynb</a></td>
<td><a href="https://youtu.be/cNADlHfHQHg">YouTube</a></td>
</tr>
<tr>
<td>9/12</td>
<td>6 - Optimization</td>
<td>Kolter</td>
<td><a href="https://dlsyscourse.org/slides/fc_init_opt.pdf">pdf</a></td>
<td><a href="https://youtu.be/CukpVt-1PA4">YouTube</a></td>
</tr>
<tr>
<td>9/17</td>
<td>7 - Neural Network Library Abstractions</td>
<td>Chen</td>
<td><a href="https://dlsyscourse.org/slides/7-nn-framework.pdf">pdf</a></td>
<td><a href="https://youtu.be/fzKNkS_5E6U">YouTube</a></td>
</tr>
<tr>
<td>9/19</td>
<td>9 - Normalization, Dropout, + Implementation</td>
<td>Kolter</td>
<td><a href="https://dlsyscourse.org/slides/norm_reg.pdf">pdf</a></td>
<td><a href="https://youtu.be/ky7qiKyZmnE">YouTube</a></td>
</tr>
<tr>
<td>9/24</td>
<td>8 - NN Library Implementation</td>
<td>Chen</td>
<td><a href="https://github.com/dlsyscourse/lecture8/blob/main/8_nn_library_implementation.ipynb">ipynb</a></td>
<td><a href="https://youtu.be/uB81vGRrH0c">YouTube</a></td>
</tr>
<tr>
<td>9/26</td>
<td>10 - Convolutional Networks</td>
<td>Kolter</td>
<td><a href="https://dlsyscourse.org/slides/conv_nets.pdf">pdf</a></td>
<td><a href="https://youtu.be/-5RPPjn0hPg">YouTube</a></td>
</tr>
<tr>
<td>10/1</td>
<td>11 - Hardware Acceleration for Linear Algebra</td>
<td>Chen</td>
<td><a href="https://dlsyscourse.org/slides/11-hardware-acceleration.pdf">pdf</a></td>
<td><a href="https://youtu.be/es6s6T1bTtI">YouTube</a></td>
</tr>
<tr>
<td>10/3</td>
<td>12 - Hardware Acceleration + GPUs</td>
<td>Chen</td>
<td><a href="https://dlsyscourse.org/slides/12-gpu-acceleration.pdf">pdf</a></td>
<td><a href="https://youtu.be/jYCxVirq4d0">YouTube</a></td>
</tr>
<tr>
<td>10/8</td>
<td>13 - Hardware Acceleration Implementation</td>
<td>Chen</td>
<td><a href="https://github.com/dlsyscourse/lecture13/blob/main/13_hardware_acceleration_architecture_overview.ipynb">ipynb</a></td>
<td><a href="https://youtu.be/XdhUZRXA7fg">YouTube</a></td>
</tr>
<tr>
<td>10/10</td>
<td>14 - Convoluations Network Implementation</td>
<td>Kolter</td>
<td><a href="https://github.com/dlsyscourse/public_notebooks/blob/main/convolution_implementation.ipynb">ipynb</a></td>
<td><a href="https://youtu.be/7kclgMIcMq0">YouTube</a></td>
</tr>
<tr>
<td>10/15</td>
<td><em>No class - Fall Break</em></td>
<td> </td>
<td> </td>
<td> </td>
</tr>
<tr>
<td>10/17</td>
<td><em>No class - Fall Break</em></td>
<td> </td>
<td> </td>
<td> </td>
</tr>
<tr>
<td>10/22</td>
<td>15 - Sequence Modeling + RNNs</td>
<td>Kolter</td>
<td><a href="https://dlsyscourse.org/slides/rnns.pdf">pdf</a></td>
<td><a href="https://youtu.be/aI47BqLYahc">YouTube</a></td>
</tr>
<tr>
<td>10/24</td>
<td>16 - Sequence Modeling Implementation</td>
<td>Kolter</td>
<td><a href="https://github.com/dlsyscourse/public_notebooks/blob/main/rnn_implementation.ipynb">ipynb</a></td>
<td><a href="https://youtu.be/q12VPh-bK7k">YouTube</a></td>
</tr>
<tr>
<td>10/29</td>
<td>17 - Transformers and Autoregressive Models</td>
<td>Kolter</td>
<td><a href="https://dlsyscourse.org/slides/transformers.pdf">pdf</a></td>
<td><a href="https://youtu.be/IFKRf-BAqZo">Youtube</a></td>
</tr>
<tr>
<td>10/31</td>
<td>18 - Transformers Implementation</td>
<td>Kolter</td>
<td><a href="https://github.com/dlsyscourse/public_notebooks/blob/main/transformer_implementation.ipynb">ipynb</a></td>
<td><a href="https://youtu.be/OzFmKdAHJn0">Youtube</a></td>
</tr>
<tr>
<td>11/5</td>
<td><em>No class - Democracy Day</em></td>
<td> </td>
<td> </td>
<td> </td>
</tr>
<tr>
<td>11/7</td>
<td>19 - Training Large Models</td>
<td>Chen</td>
<td><a href="https://dlsyscourse.org/slides/15-training-large-models.pdf">pdf</a></td>
<td><a href="https://youtu.be/HSzVogM5IPo">YouTube</a></td>
</tr>
<tr>
<td>11/12</td>
<td>20 - Generative Models</td>
<td>Chen</td>
<td><a href="https://dlsyscourse.org/slides/16-generative-models.pdf">pdf</a></td>
<td><a href="https://youtu.be/iIx_8_pxzhs">YouTube</a></td>
</tr>
<tr>
<td>11/14</td>
<td>21 - Generative Models Implementation</td>
<td>Chen</td>
<td><a href="https://github.com/dlsyscourse/public_notebooks/blob/main/21_generative_adversarial_networks_implementation.ipynb">ipynb</a></td>
<td><a href="https://youtu.be/DmBw8SEeAc0">YouTube</a></td>
</tr>
<tr>
<td>11/19</td>
<td>22 - Customize Pretrained Models</td>
<td>Chen</td>
<td><a href="https://dlsyscourse.org/slides/22-augment-pretrained-models.pdf">pdf</a></td>
<td> </td>
</tr>
<tr>
<td>11/21</td>
<td>23 - Model Deployment</td>
<td>Chen</td>
<td><a href="https://dlsyscourse.org/slides/23-model-deployment.pdf">pdf</a></td>
<td><a href="https://youtu.be/jCBrUisBQ0A">Youtube</a></td>
</tr>
<tr>
<td>11/26</td>
<td>24 - Machine Learning Compilation and Deployment Implementation</td>
<td>Chen</td>
<td><a href="https://github.com/dlsyscourse/public_notebooks/blob/main/24_machine_learning_compilation_deployment_implementation.ipynb">ipynb</a></td>
<td><a href="https://youtu.be/HIwsCzdW_pw">Youtube</a></td>
</tr>
<tr>
<td>11/28</td>
<td><em>No class - Thanksgiving</em></td>
<td> </td>
<td> </td>
<td> </td>
</tr>
<tr>
<td>12/3</td>
<td>25 - Future Directions / Q&amp;A</td>
<td>Both</td>
<td> </td>
<td> </td>
</tr>
<tr>
<td>12/5</td>
<td>26 - Student project presentations</td>
<td>Students</td>
<td> </td>
<td> </td>
</tr>
</tbody>
</div></div></div><br>
<br>
<br>
<br>
<br>
<br>
via DL Systems <br>
<br>
May 22, 2025 at 02:04PM
